asset-transfer-basic/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,246 @@
+    'msp',
+    'signcerts',
+    'Admin@org1.example.com-cert.pem',
+);
+//path to peer tls certificate
+const tlsCertPath = path.resolve(
+    cryptoPath,
+    'peers',
+    'peer0.org1.example.com',
+    'tls',
+    'ca.crt',
+);
+
+//Gateway peer endpoint
+const peerEndpoint = 'localhost:7051';
+
+// pre-requisites:
+// - fabric-sample two organization test-network setup with two peers and ordering service
+//         ===> from directory /fabric-samples/test-network
+//         ./network.sh up createChannel
+// - Use any of the asset-transfer-basic chaincodes deployed on the channel "mychannel"
+//   with the chaincode name of "basic". The following deploy command will package,
+//   install, approve, and commit the javascript chaincode, all the actions it takes
+//   to deploy a chaincode to a channel.
+//         ===> from directory /fabric-samples/test-network
+//         ./network.sh deployCC -ccn basic -ccp ../asset-transfer-basic/chaincode-typescript/ -ccl typescript
+// - Be sure that node.js is installed
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-typescript
+//         node -v
+// - npm installed code dependencies
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-typescript
+//         npm install
+// - to run this test application
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-typescript
+//         npm start
+
+/**
+ *  A test application to show basic queries operations with any of the asset-transfer-basic chaincodes
+ *   -- How to submit a transaction
+ *   -- How to query and check the results
+ *
+ * To see the SDK workings, try setting the logging to show on the console before running
+ *        export HFC_LOGGING='{"debug":"console"}'
+ */
+
+async function main():Promise<void> {
+
+    // The gRPC client connection should be shared by all Gateway connections to this endpoint
+    const client = await newGrpcConnection();
+
+    const gateway = connect({
+        client,
+        identity: await newIdentity(),
+        signer: await newSigner(),
+    });
+
+    try {
+
+        // Build a network instance based on the channel where the smart contract is deployed
+        const network = gateway.getNetwork(channelName);
+
+        // Get the contract from the network.
+        const contract = network.getContract(chaincodeName);
+
+        // Initialize a set of asset data on the channel using the chaincode 'InitLedger' function.
+        await initLedger(contract);
+
+        //Return all the current assets on the ledger.
+        await getAllAssets(contract);
+
+        //Create new asset on the ledger.
+        await createAsset(contract);
+
+        //Update an existing asset asynchronously.
+        await updateAssetAsync(contract);
+
+        //Get the asset details by assetID
+        await readAssetByID(contract);
+
+        //Update an asset which does not exist.
+        await updateAsset(contract)
```

Comment:
It's a really small niggle but, for clarity, the function name could either reflect that it's going to fail, or that it's updating a non-existent asset